executor/show.go

```diff
@@ -843,6 +845,29 @@ func (e *ShowExec) fetchShowCollation() error {
 	return nil
 }
 
+// fetchShowCreateUser composes show create create user result.
+func (e *ShowExec) fetchShowCreateUser() error {
+	checker := privilege.GetPrivilegeManager(e.ctx)
+	if checker == nil {
+		return errors.New("miss privilege checker")
+	}
+	sql := fmt.Sprintf(`SELECT * FROM %s.%s WHERE User='%s' AND Host='%s';`,
+		mysql.SystemDB, mysql.UserTable, e.User.Username, e.User.Hostname)
+	rows, _, err := e.ctx.(sqlexec.RestrictedSQLExecutor).ExecRestrictedSQL(e.ctx, sql)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if len(rows) == 0 {
+		return ErrCannotUser.GenWithStackByArgs("SHOW CREATE USER",
```

Comment:
This is the wrong error code for MySQL compatibility.  It should be:
```
return ErrNonexistingGrant.GenWithStackByArgs(e.User.Username, e.User.Hostname)
```
You will need to add this error to `./executor/errors.go`:
```
@@ -48,6 +48,7 @@ var (
        ErrDBaccessDenied              = terror.ClassExecutor.New(mysql.ErrDBaccessDenied, mysql.MySQLErrName[mysql.ErrDBaccessDenied])
        ErrTableaccessDenied           = terror.ClassExecutor.New(mysql.ErrTableaccessDenied, mysql.MySQLErrName[mysql.ErrTableaccessDenied])
        ErrBadDB                       = terror.ClassExecutor.New(mysql.ErrBadDB, mysql.MySQLErrName[mysql.ErrBadDB])
+       ErrNonexistingGrant            = terror.ClassExecutor.New(mysql.ErrNonexistingGrant, mysql.MySQLErrName[mysql.ErrNonexistingGrant])
 )
 
 func init() {
@@ -63,6 +64,7 @@ func init() {
                mysql.ErrDBaccessDenied:              mysql.ErrDBaccessDenied,
                mysql.ErrTableaccessDenied:           mysql.ErrTableaccessDenied,
                mysql.ErrBadDB:                       mysql.ErrBadDB,
+               mysql.ErrNonexistingGrant:            mysql.ErrNonexistingGrant,
        }
        terror.ErrClassToMySQLCodes[terror.ClassExecutor] = tableMySQLErrCodes
 }
```

Reply:
This is what I see in mysql of version 5.7.17. What is your mysql version?

```
mysql: [Warning] Using a password on the command line interface can be insecure.
Welcome to the MySQL monitor.  Commands end with ; or \g.
Your MySQL connection id is 6
Server version: 5.7.17 Homebrew

Copyright (c) 2000, 2016, Oracle and/or its affiliates. All rights reserved.

Oracle is a registered trademark of Oracle Corporation and/or its
affiliates. Other names may be trademarks of their respective
owners.

Type 'help;' or '\h' for help. Type '\c' to clear the current input statement.

mysql> show create user 'aaa'@'aaa';
ERROR 1396 (HY000): Operation SHOW CREATE USER failed for 'aaa'@'aaa'

```

Reply:
@lnhote I checked 5.7 again, you are correct :-)  Sorry about that.